Modelica/Utilities/Internal.mo

```diff
@@ -240,4 +240,233 @@ especially if the operating system supports Unicode characters.
+      output Integer min "Minute";
+      output Integer hour "Hour";
+      output Integer day "Day";
+      output Integer mon "Month";
+      output Integer year "Year";
+      external "C" ModelicaInternal_getTime(ms, sec, min, hour, day, mon, year)
+        annotation(Library="ModelicaExternalC");
+      annotation (Documentation(info="<html>
+<h4>Syntax</h4>
+<blockquote><pre>
+(ms, sec, min, hour, day, mon, year) = Internal.Time.<strong>getTime</strong>();
+</pre></blockquote>
+<h4>Description</h4>
+<p>
+Returns the local time at the time instant this function was called.
+All returned values are of type Integer and have the following meaning:
+</p>
+
+<blockquote>
+<table border=\"1\" cellspacing=\"0\" cellpadding=\"2\">
+<tr><th>Argument</th>
+    <th>Range</th>
+    <th>Description</th></tr>
+
+<tr><td>ms</td> <td>0&nbsp;&hellip;&nbsp;999</td>
+    <td>Milli-seconds after seconds</td></tr>
+
+<tr><td>sec</td> <td>0&nbsp;&hellip;&nbsp;59</td>
+    <td>Seconds after minute</td></tr>
+
+<tr><td>min</td> <td>0&nbsp;&hellip;&nbsp;59</td>
+    <td>Minutes after hour</td></tr>
+
+<tr><td>hour</td> <td>0&nbsp;&hellip;&nbsp;23</td>
+    <td>Hours after midnight</td></tr>
+
+<tr><td>day</td> <td>1&nbsp;&hellip;&nbsp;31</td>
+    <td>Day of month</td></tr>
+
+<tr><td>mon</td> <td>1&nbsp;&hellip;&nbsp;12</td>
+    <td>Current month</td></tr>
+
+<tr><td>year</td> <td>&ge; 1970</td>
+    <td>Current year</td></tr>
+</table>
+</blockquote>
+
+<h4>Example</h4>
+<blockquote><pre>
+(ms, sec, min, hour, mon, year) = getTime()   // = (281, 30, 13, 10, 15, 2, 2015)
+                                              // Feb. 15, 2015 at 10:13 after 30.281 s
+</pre></blockquote>
+<h4>Note</h4>
+<p>This function is impure!</p>
+</html>",   revisions="<html>
+<table border=\"1\" cellspacing=\"0\" cellpadding=\"2\">
+<tr><th>Date</th> <th align=\"left\">Description</th></tr>
+
+<tr><td> June 22, 2015 </td>
+    <td>
+
+<table border=\"0\">
+<tr><td>
+         <img src=\"modelica://Modelica/Resources/Images/Logos/dlr_logo.png\">
+</td><td valign=\"bottom\">
+         Initial version implemented by
+         A. Kl&ouml;ckner, F. v.d. Linden, D. Zimmer, M. Otter.<br>
+         <a href=\"http://www.dlr.de/rmc/sr/en\">DLR Institute of System Dynamics and Control</a>
+</td></tr></table>
+</td></tr>
+
+</table>
+</html>"));
+    end getTime;
+
+    pure function localTime "Retrieve the local time (in the local time zone) from elapsed seconds since reference year"
+      extends Modelica.Icons.Function;
+      input Real seconds "Elapsed seconds since epoch_year";
+      input Integer epoch_year = 1970 "Reference year";
+      output Integer ms "Millisecond";
+      output Integer sec "Second";
+      output Integer min "Minute";
+      output Integer hour "Hour";
+      output Integer day "Day";
+      output Integer mon "Month";
+      output Integer year "Year";
+      external "C90" ModelicaTime_localtime(ms, sec, min, hour, day, mon, year, seconds, epoch_year)
```

Comment:
@HansOlsson I explicitly set "C90" here since the C implementation uses modf (http://www.cplusplus.com/reference/cmath/modf/) of C90. OK for you?

Reply:
Sure.

Reply:
OK.